pandas/io/parsers/base_parser.py

```diff
@@ -120,13 +120,7 @@ def __init__(self, kwds) -> None:
 
         # validate header options for mi
         self.header = kwds.get("header")
-        if isinstance(self.header, (list, tuple, np.ndarray)):
-            if not all(map(is_integer, self.header)):
-                raise ValueError("header must be integer or list of integers")
-            if any(i < 0 for i in self.header):
-                raise ValueError(
-                    "cannot specify multi-index header with negative integers"
-                )
```

Comment:
I might be missing it, is validate_header_arg called somewhere instead?

Reply:
Yes, but it took me a while to find it. It's called earlier in TextFileReader:

```
  pandas/io/parsers/readers.py(1847)TextParser()
-> return TextFileReader(*args, **kwds)

  pandas/io/parsers/readers.py(1412)__init__()
-> self.options, self.engine = self._clean_options(options, engine)

  pandas/io/parsers/readers.py(1607)_clean_options()
-> validate_header_arg(options["header"])
```